engine/src/types.rs

```diff
@@ -196,6 +199,64 @@ macro_rules! declare_types {
     };
 }
 
+/// type Array<'a> = Vec<LhsValue<'a>>;
```

Comment:
Maybe a comment for similar to the one for `Map`? Something like ` /// An array of [`Type`].`